Fix deprecated warning caused by shuffle_and_repeat when using make_csv_dataset

yongtang opened the pull request:
This fix fixes unnecessary deprecated warning caused by shuffle_and_repeat when using make_csv_dataset:
```
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow_core/python/data/experimental/ops/readers.py:215: shuffle_and_repeat (from tensorflow.python.data.experimental.ops.shuffle_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.Dataset.shuffle(buffer_size, seed)` followed by `tf.data.Dataset.repeat(count)`. Static tf.data optimizations will take care of using the fused implementation.
```

Note the fix doesn't use shuffle().repeat() as was suggested in the warning message, instead this fix keeps the original implementation to avoid potential performance issues.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix deprecated warning caused by shuffle_and_repeat when using make_csv_dataset

This fix fixes deprecated warning caused by shuffle_and_repeat when using make_csv_dataset:
```
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow_core/python/data/experimental/ops/readers.py:215: shuffle_and_repeat (from tensorflow.python.data.experimental.ops.shuffle_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.Dataset.shuffle(buffer_size, seed)` followed by `tf.data.Dataset.repeat(count)`. Static tf.data optimizations will take care of using the fused implementation.
```

Note the fix keeps the original implementation (vs.
using shuffle().repeat() as was suggested in the warning message) to avoid potential performance
issues

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

jsimsa commented on tensorflow/python/data/experimental/ops/readers.py:
```diff
@@ -209,9 +209,10 @@ def _maybe_shuffle_and_repeat(
-    return dataset.apply(
-        shuffle_ops.shuffle_and_repeat(shuffle_buffer_size, num_epochs,
-                                       shuffle_seed))
+    def _apply_fn(input_dataset):  # pylint: disable=missing-docstring
```
you can just return `shuffle_ops._ShuffleAndRepeatDataset(...)` instead of doing the `apply`

jsimsa replied:
and for consistency, it might make sense to do the same for `parallel_interleave` (instead of the `_parallel_interleave` indirection you introduced)

yongtang committed:
Change to use shuffle_ops._ShuffleAndRepeatDataset directly

based on review feedback

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Use core_readers.ParallelInterleaveDataset to replace interleave_ops._parallel_interleave

from feedback in review

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Remove not needed interleave_ops import

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

jsimsa commented on tensorflow/python/data/experimental/ops/interleave_ops.py:
```diff
@@ -105,8 +88,12 @@ def parallel_interleave(map_func,
-  return _parallel_interleave(map_func, cycle_length, block_length, sloppy,
-                              buffer_output_elements, prefetch_input_elements)
+  def _apply_fn(dataset):
+    return readers.ParallelInterleaveDataset(
```
you can just return readers.ParallelInterleaveDataset here

yongtang committed:
Fix test failure

filename_to_dataset should be changed to lambda filename: reader(filename, *reader_args),

(copy-past-error)

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @jsimsa for the review. There was a test failure (passed a typo name). I have updated the PR and pushed. Can you look and approve again if everything is still fine? Sorry for the extra inconvenience.
